fedimint-api/src/hex/serde.rs

```diff
@@ -0,0 +1,53 @@
+use std::borrow::Cow;
+
+use bitcoin_hashes::hex::{FromHex, ToHex};
+use serde::de::Error;
+use serde::{Deserialize, Serializer};
+
+/// Serialize a `Vec<u8>` to a hex String
+pub fn serialize<S>(data: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let hex: String = data.to_hex();
+    serializer.serialize_str(&hex)
+}
+
+/// Deserialize a hex String to a `Vec<u8>`
+pub fn deserialize<'de, D: serde::Deserializer<'de>>(d: D) -> Result<Vec<u8>, D::Error> {
+    let hex_bytes: Cow<'de, str> = Deserialize::deserialize(d)?;
```

Comment:
I thought about this lifetime here for `Cow<'de, str>`. It works with and without `'de`. I think it's fine to attach the lifetime of the Cow to the deserializer here?